### ISSUE TITLE
spaces in variable names not show chart

### ISSUE THREAD

**Issue by mattijn** [altair-viz/altair#1353]
No idea what happened, but if I have spaces in variable names the generated chart doesn't include data anymore:
![image](https://user-images.githubusercontent.com/5186265/53648305-d70bed00-3c3f-11e9-8c91-7851029b9db2.png)

And the generated vega-lite spec becomes as follow:
```json
{
  "config": {"view": {"width": 400, "height": 300}},
  "data": {"name": "data-35b30af4ed30f135cab3ab53fa6c019d"},
  "mark": "bar",
  "encoding": {
    "x": {"type": "nominal", "field": "a space"},
    "y": {"type": "quantitative", "field": "b space"}
  },
  "height": 200,
  "$schema": "https://vega.github.io/schema/vega-lite/v2.6.0.json",
  "datasets": {
    "data-35b30af4ed30f135cab3ab53fa6c019d": [
      {"_0": "A", "_1": 28},
      {"_0": "B", "_1": 55},
      {"_0": "C", "_1": 43},
      {"_0": "D", "_1": 91},
      {"_0": "E", "_1": 81},
      {"_0": "F", "_1": 53},
      {"_0": "G", "_1": 19},
      {"_0": "H", "_1": 87},
      {"_0": "I", "_1": 52}
    ]
  }
}
```
I've seen this issue https://github.com/altair-viz/altair/issues/161, which sounds similar, but is from a long time ago. 

I tried with both altair version 2.4.1 and master version, but with similar behavior.
Used altair in screenshot:
```python
import altair as alt
import pandas as pd

source = pd.DataFrame({
    'a space': ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I'],
    'b space': [28, 55, 43, 91, 81, 53, 19, 87, 52]
})

alt.Chart(source, height=200).mark_bar().encode(
    x='a space',
    y='b space'
)
```

**Comment by jakevdp** [comment]
I can't reproduce this issue. I ran your code snippet in JupyterLab with the current Altair version, and saw the expected output:
![visualization 44](https://user-images.githubusercontent.com/781659/53648813-a901bb00-3bf5-11e9-9682-eddabd847570.png)

**Comment by jakevdp** [comment]
I'm most confused about how your data got rewritten to columns named ``"_0"`` and ``"_1"``. The only possibility I can think of is that you've somehow enabled a custom data transformer that does strange things with the names. What's the output of the following for you?
```python
>>> alt.data_transformers
DataTransformerRegistry(active='default', registered=['csv', 'default', 'json'])
```

**Comment by mattijn** [comment]
```python
>>> alt.data_transformers
DataTransformerRegistry(active='default', registered=['csv', 'data_server', 'default', 'json'])
```
one difference is the `data_server`

**Comment by jakevdp** [comment]
There's nothing I know in Altair that could change column names from ``['a space', 'b space']`` to ``['_0', '_1']``, and whatever is doing that is the root of the problem.

What is the output of ``source.to_json()``?

**Comment by mattijn** [comment]
```python
>>> source.to_json()
'{"a space":{"0":"A","1":"B","2":"C","3":"D","4":"E","5":"F","6":"G","7":"H","8":"I"},"b space":{"0":28,"1":55,"2":43,"3":91,"4":81,"5":53,"6":19,"7":87,"8":52}}'
```

**Comment by jakevdp** [comment]
How about ``source.to_dict(orient='records')`` ?

**Comment by jakevdp** [comment]
Also please check the output of this:
```python
from altair.utils.core import sanitize_dataframe
sanitize_dataframe(source)
```

**Comment by jakevdp** [comment]
And this:
```python
transformer = alt.data_transformers.get()
transformer(source)
```

**Comment by mattijn** [comment]
```python
>>> source.to_dict(orient='records')
[{'_0': 'A', '_1': 28},
 {'_0': 'B', '_1': 55},
 {'_0': 'C', '_1': 43},
 {'_0': 'D', '_1': 91},
 {'_0': 'E', '_1': 81},
 {'_0': 'F', '_1': 53},
 {'_0': 'G', '_1': 19},
 {'_0': 'H', '_1': 87},
 {'_0': 'I', '_1': 52}]
```
That's a pandas function.. (my pandas version is `0.24.0`)

`sanitize_dateframe` is giving correct output and the last is wrong again

```python
>>> transformer = alt.data_transformers.get()
>>> transformer(source)
{'values': [{'_0': 'A', '_1': 28},
  {'_0': 'B', '_1': 55},
  {'_0': 'C', '_1': 43},
  {'_0': 'D', '_1': 91},
  {'_0': 'E', '_1': 81},
  {'_0': 'F', '_1': 53},
  {'_0': 'G', '_1': 19},
  {'_0': 'H', '_1': 87},
  {'_0': 'I', '_1': 52}]}
```

**Comment by jakevdp** [comment]
OK, so it looks like the issue is that pandas is rewriting your column names when converting the dataframe to dict. I wonder if there was some change introduced in Pandas 0.24?

**Comment by jakevdp** [comment]
I've tried on Python 2.7 and Python 3, with pandas 0.22 and 0.24, and cannot reproduce those ``to_dict()`` output with any of the four combinations. Any ideas?

**Comment by jakevdp** [comment]
Just to be clear, this is the output I expect, and that I get in every environment I try:
```python
In [1]: import pandas as pd

In [2]: source = pd.DataFrame({
   ...:     'a space': ['A', 'B', 'C'],
   ...:     'b space': [28, 55, 43]
   ...: })

In [3]: source.to_dict(orient='records')
Out[3]: 
[{'a space': 'A', 'b space': 28},
 {'a space': 'B', 'b space': 55},
 {'a space': 'C', 'b space': 43}]
```

**Comment by jakevdp** [comment]
Maybe related to this: https://stackoverflow.com/questions/54531367/pandas-dataframe-to-dict-behavior

**Comment by jakevdp** [comment]
I would open an issue on the pandas bug tracker with the code snippet from https://github.com/altair-viz/altair/issues/1353#issuecomment-468732670, noting your exact python version and pandas version.

**Comment by jakevdp** [comment]
Ah, here's the relevant github issue: https://github.com/pandas-dev/pandas/issues/25078

**Comment by mattijn** [comment]
Should be a bug somewhere in pandas then. That question of SO is the same whats happening here. But I can't remember that I update pandas lately and I can't imagine that I've been working with solely variables without spaces in altair.
Confused.

**Comment by mattijn** [comment]
Ah, ok, seems to be fixed already! Then will close this issue as well!

**Comment by jakevdp** [comment]
Looks like pandas 0.24.1 fixes the regression that was introduced in 0.24.0
